jpa/subsystem/src/main/java/org/jboss/as/jpa/classloader/JPADelegatingClassFileTransformer.java

```diff
@@ -24,33 +24,68 @@
 
 import java.lang.instrument.ClassFileTransformer;
 import java.lang.instrument.IllegalClassFormatException;
+import java.nio.ByteBuffer;
 import java.security.ProtectionDomain;
 
+import org.jboss.as.jpa.messages.JpaLogger;
 import org.jipijapa.plugin.spi.PersistenceUnitMetadata;
 
 /**
  * Helps implement PersistenceUnitInfo.addClassTransformer() by using DelegatingClassFileTransformer
  *
  * @author Scott Marlow
  */
-public class JPADelegatingClassFileTransformer implements ClassFileTransformer {
+public class JPADelegatingClassFileTransformer implements ClassFileTransformer, org.jboss.modules.ClassTransformer {
     private final PersistenceUnitMetadata persistenceUnitMetadata;
 
     public JPADelegatingClassFileTransformer(PersistenceUnitMetadata pu) {
         persistenceUnitMetadata = pu;
     }
 
     @Override
-    public byte[] transform(ClassLoader classLoader, String className, Class<?> aClass, ProtectionDomain protectionDomain, byte[] originalBuffer) throws
-        IllegalClassFormatException {
-        byte[] transformedBuffer = originalBuffer;
+    public byte[] transform(ClassLoader classLoader, String className, Class<?> aClass, ProtectionDomain protectionDomain, byte[] originalBuffer) {
+        return getBytes(transform(classLoader, className, protectionDomain, ByteBuffer.wrap(originalBuffer)));
+    }
+
+    @Override
+    public ByteBuffer transform(ClassLoader classLoader, String className, ProtectionDomain protectionDomain, ByteBuffer classBytes)
+            throws IllegalArgumentException {
+        final byte[] bytes = getBytes(classBytes);
+        final Class<?> aClass = null;
+        boolean transformed = false;
+        byte[] transformedBuffer = bytes;
```

Comment:
Having both `bytes` and `transformedBuffer` is redundant.  You can just use `bytes` directly and drop `transformedBuffer`.

Reply:
true, will drop bytes, as I think we need transformedBuffer.